### PR TITLE
feat(devserver): Force latest offset in devserver

### PR DESCRIPTION
**Pull request by mitsuhiko**
Currently if you don't run the devservices continuously before the topic gets too old you will have a stuck snuba. Right now the workflow is to recreate the underlying volumes but that seems pretty silly.

There does not seem to be much of a disadvantage to forcing the offset to latest. In fact this might even make it a more sensible setup as you will not have to chug through old alerts and similar nonsense in dev.

Requires https://github.com/getsentry/arroyo/pull/54

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,7 @@ backend-typing:
 install-python-dependencies:
 	pip install -e .
 	pip install -r requirements-test.txt
+	pip install git+https://github.com/getsentry/arroyo@0f743510370fa933595c3dc13390b6a6b84551a8#egg=sentry-arroyo
```
Comment (mitsuhiko) [inline Makefile:27]: TODO: remove this and bump arroyo in requirements instead.

File: Makefile [hunk 2]
```diff
@@ -24,6 +24,7 @@ backend-typing:
 install-python-dependencies:
 	pip install -e .
 	pip install -r requirements-test.txt
+	pip install git+https://github.com/getsentry/arroyo@d4caa8d3b267c30efb5a996545cdd0e5ce5cb7b2#egg=sentry-arroyo
```
Comment (mitsuhiko) [inline Makefile:27]: This needs undoing before merge with a bump to a new arroyo release.

Comment (rahul-kumar-saini) [reply Makefile:29]: This should unblock:
https://github.com/getsentry/snuba/pull/2670

File: snuba/consumers/strict_consumer.py [hunk 3]
```diff
@@ -64,6 +64,10 @@ def __init__(
         self.__topic = topic
         self.__consuming = False
 
+        # This is a made up flag that exists in arroyo.  This consumer always behaves
+        # as if it forces the offset reset so remove it.
+        broker_config.pop("force.offset.reset", None)
```
Comment (lynnagara) [inline snuba/consumers/strict_consumer.py:69]: Didn't this end up being `arroyo.strict.offset.reset`?

File: snuba/subscriptions/consumer.py [hunk 4]
```diff
@@ -193,8 +193,10 @@ def seek(self, offsets: Mapping[Partition, int]) -> None:
 
         return self.__consumer.seek(offsets)
 
-    def stage_positions(self, positions: Mapping[Partition, Position]) -> None:
-        return self.__consumer.stage_positions(positions)
+    def stage_positions(
+        self, positions: Mapping[Partition, Position], force: bool = False
```
Comment (lynnagara) [inline snuba/subscriptions/consumer.py:197]: You can remove force now

**Review by lynnagara — APPROVED** [review]
Thanks, will make a lot of people happy

**Pull request merged**